src/worker/ingestion/process-event.ts

```diff
@@ -192,11 +192,46 @@ export class EventsProcessor {
         if (event === '$create_alias') {
             await this.alias(properties['alias'], distinctId, teamId)
         } else if (event === '$identify') {
-            if (properties['$anon_distinct_id']) {
-                await this.alias(properties['$anon_distinct_id'], distinctId, teamId)
+            await this.handleIdentify(event, properties, distinctId, teamId)
+        }
+    }
+
+    private async handleIdentify(
+        event: string,
+        properties: Properties,
+        distinctId: string,
+        teamId: number
+    ): Promise<void> {
+        // Ensure that if the person identified by "$anon_distinct_id" is
+        // identified, that we do not alias to `distinctId`, as this would
+        // be merging two distinct identified persons. See
+        // https://github.com/PostHog/posthog/issues/5527 for discussion on
+        // the details of this guard is protecting against
+        const previousDistinctId = properties['$anon_distinct_id']
+        if (previousDistinctId) {
+            const previousPerson = await this.db.fetchPerson(teamId, previousDistinctId)
```

Comment:
Why did we go back down to this route? :thinking: 

Couple of issues:
1. We're losing the property that "only anonymous users get merged"
2. We're increasing load on the DB due to repeated lookups

Worth discussing face-to-face?

Reply:
I wasn't sure that "only anonymous users get merged" was a requirement but will update accordingly.